substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationManagementExt.java

```diff
@@ -77,7 +77,8 @@ private void linkManagementExt(@SuppressWarnings("unused") DuringAnalysisAccess
 
     @Override
     public void afterAnalysis(AfterAnalysisAccess access) {
-        if (NativeLibrarySupport.singleton().isPreregisteredBuiltinLibrary("awt_headless")) {
+        boolean managementExtNeeded = nativeLibraries.getStaticLibraries().stream().anyMatch(l -> l.equals(nativeLibraries.getStaticLibraryPath("management_ext")));
```

Comment:
I think it would be simpler to just check if the `com.sun.management.internal.OperatingSystemImpl` class is reachable.
```suggestion
        boolean managementExtNeeded = access.isReachable(clazz(access, "com.sun.management.internal.OperatingSystemImpl"));
```

Reply:
Thanks, done in the latest version.